eif_utils/src/lib.rs

```diff
@@ -519,15 +669,71 @@ impl<T: Digest + Debug + Write + Clone> EifBuilder<T> {
         if self.sign_info.is_some() {
             self.generate_eif_signature(&measurements);
         }
+        self.save_metadata().expect("Failed to save metadata");
         self.compute_crc();
         self.write_header(output_file);
         self.write_kernel(output_file);
         self.write_cmdline(output_file);
         self.write_ramdisks(output_file);
         self.write_signature(output_file);
+        self.write_metadata(output_file);
         measurements
     }
 
+    pub fn get_generated_meta(&mut self) -> BTreeMap<String, String> {
+        let mut meta = BTreeMap::new();
+
+        let time = SystemTime::now();
+        let time: DateTime<Utc> = time.into();
+        meta.insert("BuildTime".to_string(), time.to_rfc3339());
+
+        let blobs_path =
+            std::env::var("NITRO_CLI_BLOBS").unwrap_or_else(|_| DEFAULT_BLOBS_PATH.to_string());
+        // Read aws-nitro-enclaves-cli.spec file and parse the lines that contain:
+        // Name:      aws-nitro-enclaves-cli
+        // Version:   1.0.12
+        let specs_path =
+            std::env::var("NITRO_CLI_SPECS").unwrap_or_else(|_| DEFAULT_SPECS_PATH.to_string());
+        let specs_file = File::open(format!("{}/aws-nitro-enclaves-cli.spec", specs_path))
+            .expect("Failed to open NitroCLI specs file.");
+        let spec_lines = BufReader::new(specs_file).lines();
+        let mut tool_name = String::new();
+        let mut tool_version = String::new();
+
+        for line in spec_lines {
+            let line = line.expect("Failed to extract line from specs");
+            let clean_line = line.replace(' ', "");
```

Comment:
Nit: This may not work as expected if the file contains tabs instead of blanks.

Reply:
outdated